modules/library/cql/src/main/java/org/geotools/filter/text/commons/BuildResultStack.java

```diff
@@ -185,11 +185,11 @@ public double popDoubleValue() throws CQLException {
     public int popIntegerValue() throws CQLException {
         try {
             Literal expr = this.popLiteral();
-            Integer number = (Integer) expr.getValue();
+            Number number = (Number) expr.getValue();
 
             return number.intValue();
         } catch (ClassCastException cce) {
-            throw new CQLException("Expected double");
+            throw new CQLException("Expected integer");
```

Comment:
Clearly the exception message mention of "double" was wrong, but why is the type now `Number` not `Integer`?

Reply:
Because the code in question was apparently never used, and the parsing actually emits a Long instead of a Integer. Since I was not sure, I switched to the base class for extra safety.

Reply:
OK. The exception makes sense when understood as little-i integer, the category not a data type.